### ISSUE TITLE
Annotate the syntax tree with the information to extract the number of constant pi groups

### ISSUE THREAD

**Issue by KisekiHirakawa** [phillipstanleymarbell/Noisy-lang-compiler#442]
**Is your feature request related to a problem?**
Currently, we obtain the list of identifiers from the invariant body(for constant handling). We then annotate the dimensional matrix from this list and work out the number of constant pi groups from the AST. We do not modify the AST at present. 

**Describe the solution you'd like**
Ideally, it would be nice to have a new member in the Invariant struct called noOfConstPi. I believe this could be done in parallel with annotating the nullSpace.
